package/MDAnalysis/core/groups.py

```diff
@@ -432,13 +498,36 @@ def __radd__(self, other):
                             " '{}' and '{}'".format(type(self).__name__,
                                                     type(other).__name__))
 
+    @_only_same_level
+    def __eq__(self, other):
+        """Test group equality.
+
+        Two groups are equal if they contain the same indices in
+        the same order. Groups that are not at the same level or that belong
+        to different universe cannot be compared.
+        """
+        o_ix = other.ix_array
```

Comment:
Here you probably need to use `.ix` otherwise Atoms evaluate equal to AtomGroups, eg `u.atoms[1] == u.atoms[[1]]`